integration/vm_factory/vm_cache_test.sh

```diff
@@ -0,0 +1,133 @@
+#!/bin/bash
+#
+# Copyright (c) 2020 ARM Limited
+#
+# SPDX-License-Identifier: Apache-2.0
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+cidir=$(dirname "$0")
+source "${cidir}/../../metrics/lib/common.bash"
+
+kata_runtime=$(command -v kata-runtime)
+
+# Environment variables
+KATA_CACHE_INIT="${kata_runtime} factory init"
+KATA_CACHE_STATUS="${kata_runtime} factory status"
+KATA_CACHE_DESTROY="${kata_runtime} factory destroy"
+KATA_CACHE_SOCK="/var/run/kata-containers/cache.sock"
+
+IMAGE="${IMAGE:-busybox}"
+TEST_PLAYLOAD="2"
+CONTAINER_NAME=$(random_name)
+PAYLOAD_ARGS="${PAYLOAD_ARGS:-tail -f /dev/null}"
+
+setup() {
+        extract_kata_env
+}
+
+teardown(){
+	IFS="", stale_process_union=("${KATA_CACHE_INIT}" "${HYPERVISOR_PATH}")
+	for stale_process in ${stale_process_union[@]}
+	do
+		local pid=$(pgrep -f "${stale_process}")
+		if [ -n "${pid}" ]; then
+			echo "${pid}" | xargs kill
+		fi
+	done
+
+	if [ -e ${KATA_CACHE_SOCK} ]; then
+		sudo sh -c 'rm -f ${KATA_CACHE_SOCK}'
+	fi
+}
+
+enable_vm_cache_config() {
+	echo "enable vm cache config ${RUNTIME_CONFIG_PATH}"
+	sudo sed -i -e 's/^#\(vm_cache_number\).*=.*$/\1 = 2/g' "${RUNTIME_CONFIG_PATH}"
+	sudo sed -i -e 's/^#\(vm_cache_endpoint =.*\)$/\1/g' "${RUNTIME_CONFIG_PATH}"
+}
+
+disable_vm_cache_config() {
+	echo "disable vm cache config ${RUNTIME_CONFIG_PATH}"
+	sudo sed -i -e 's/^\(vm_cache_number =.*\)$/#\1/g' "${RUNTIME_CONFIG_PATH}"
```

Comment:
This is not going to set `vm_cache_number` to its default value (0) but rather just comment the line. If the idea is to roll back the `configuration.toml` to its previous state, I think we should save it and copy back.